server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxSet.scala

```diff
@@ -46,7 +47,7 @@ case class MailboxSetRequest(accountId: AccountId,
                              onDestroyRemoveEmails: Option[RemoveEmailsOnDestroy]) extends WithAccountId
 
 object MailboxSetRequest {
-  type MailboxCreationId = String Refined NonEmpty
+  type MailboxCreationId = String Refined IdConstraint
```

Comment:
As far as I understand Refined, `MailboxCreationId` type is equal to `UnparsedVacationResponseId` and thus defeat the purpose of having different types. You probably need newtype here. (see https://stackoverflow.com/questions/62579057/how-to-ensure-type-safety-with-scalas-refined-library-when-using-the-same-predi for example)

Reply:
Fair. But it is not a regression, as before everything was 'refined NonEmpty'

Reply:
you are right, it's already an enhancement.

Reply:
But thanks for pointing out this good practice that was of our radar...

2 hours fighting with the code base is still acceptable, no significant rework needed so I decided to give it a shot.

We might now have some "Id POJOs" that represents similar concepts and might be de-duplicated - but I will keep that for other PRs / refactorings.